uncompyle6/semantics/make_function.py

```diff
@@ -764,52 +758,34 @@ def build_param(ast, name, default, annotation=None):
     # Unless careful, We might lose line breaks though.
     ends_in_comma = False
     if kwonlyargcount > 0:
-        if no_paramnames:
-            if not (4 & code.co_flags):
-                if argc > 0:
-                    self.write(", *, ")
-                else:
-                    self.write("*, ")
-                pass
+        if not (4 & code.co_flags):
+            if argc > 0:
+                self.write(", *, ")
             else:
-                self.write(", ")
-                ends_in_comma = True
+                self.write("*, ")
+            pass
+            ends_in_comma = True
         else:
             if argc > 0:
-                self.write(', ')
+                self.write(", ")
                 ends_in_comma = True
 
-        # FIXME: this is not correct for 3.5. or 3.6 (which works different)
-        # and 3.7?
-        if 3.0 <= self.version <= 3.2:
-            kwargs = node[0]
-            last = len(kwargs)-1
-            i = 0
-            for n in node[0]:
-                if n == 'kwarg':
-                    self.write('%s=' % n[0].attr)
-                    self.preorder(n[1])
-                    if i < last:
-                        self.write(', ')
-                        ends_in_comma = True
-                        pass
-                    else:
-                        ends_in_comma = False
-                    pass
-                i += 1
-                pass
-            pass
-        elif self.version <= 3.5:
-            # FIXME this is not qute right for 3.5
-            for n in node:
-                if n == 'pos_arg':
-                    continue
-                elif self.version >= 3.4 and not (n.kind in ('kwargs', 'no_kwargs', 'kwarg')):
-                    continue
-                else:
-                    self.preorder(n)
-                    ends_in_comma = False
-                break
+        if 3.0 <= self.version <= 3.5:
```

Comment:
I went through versions and came to this that for <= 3.5 all can be handles using following proc.
Please let me know if you have comment on this. Thanks.

Reply:
No comments other than admiration and praise. Thanks for simplifying this.